modules/core/arrow-core-data/src/main/kotlin/arrow/core/Eval.kt

```diff
@@ -271,3 +271,10 @@ sealed class Eval<out A> : EvalOf<A> {
     }
   }
 }
+
+fun <A, B> Iterator<A>.iterateRight(lb: Eval<B>): (f: (A, Eval<B>) -> Eval<B>) -> Eval<B> =
```

Comment:
Almost there! This function returns another function, which isn't necessary in Kotlin. Make `f` a second parameter instead.